proxy/src/proxy.rs

```diff
@@ -124,8 +102,19 @@ pub async fn task_main(
                     }),
                 );
             }
-            _ = PROXY_CANCELLATION_TOKEN.cancelled() => {
-                bail!("Exiting");
+            maybe_exited = connections.join_next() => {
+                match maybe_exited
+                {
+                    Some(Ok(())) => {},
+                    Some(Err(e)) => { bail!(e.to_string()) },
```

Comment:
Also for future, `bail!(e.to_string())` is never a good idea because then the caller would only have the string to work with.

`anyhow::Error::new(e).context("what happened")` works as one alternative, or in here `return Err(e.into())` would work, but it's wrong for the fn itself.

Reply:
OK, I good point. I think I didn't fully understand your code at first. I just pasted it in